Userland/Libraries/LibRegex/RegexOptimizer.cpp

```diff
@@ -36,6 +36,11 @@ void Regex<Parser>::run_optimization_passes()
     // e.g. a*b -> (ATOMIC a*)b
     attempt_rewrite_loops_as_atomic_groups(blocks);
 
+    MatchState state;
+    auto& opcode = parser_result.bytecode.get_opcode(state);
+    if (opcode.opcode_id() == OpCodeId::CheckBegin)
+        parser_result.optimization_data.only_start_of_line = true;
```

Comment:
There are a few more conditions this can be true in (e.g. within an arbitrarily nested capture group), but let's defer that to when it shows up in practice.

Reply:
I should probably add a FIXME comment, then